### ISSUE TITLE
Add ConverterCulture property on Binding to allow overriding of culture settings

### ISSUE THREAD

**Issue by jfversluis** [dotnet/maui#5696]
### Description

[This proposal was originally done for Forms](https://github.com/xamarin/Xamarin.Forms/issues/12181).

This new functionality would add the [ConverterCulture](https://docs.microsoft.com/en-us/dotnet/api/system.windows.data.binding.converterculture) property for bindings, which allows the consumer to override the locale for a binding. This functionality is also available in WPF.

### Public API Changes

Added properties:

```csharp
public class Binding
{
    // Default value would be `CurrentCulture` ideally or go to en-US.
    // The latter is what WPF does, although I'm not sure why, `CurrentCulture` makes more sense in my mind
    [System.ComponentModel.TypeConverter(typeof(CultureInfoIetfLanguageTagConverter))]
    public System.Globalization.CultureInfo ConverterCulture { get; set; }

    // Add an additional constructor to take the `ConverterCulture` parameter or create an overload for it

    // Existing Binding code here
}
```

I think we need to add our own type converter class because the one in WPF/WinUI lives in System.Windows which won't be ideal?

```csharp
// TypeConverter to make sure we can convert values like en-US and nl-NL
// to a `CultureInfo` object through XAML.
// E.g.: <Label Text="{Binding MyText, ConverterCulture='nl-NL'} />"
public class CultureInfoIetfLanguageTagConverter
{
    // Code to transform string into `CultureInfo`
}
```

### Intended Use-Case

I don't have a concrete example in mind. The proposal on the Forms repo has been upvoted a couple of times, I guess it can be useful in multi-language, multi-locale scenarios where you want to override something.

**Comment by jfversluis** [comment]
@StephaneDelcroix curious for your thoughts on this? Useful or not?

**Comment by StephaneDelcroix** [comment]
> @StephaneDelcroix curious for your thoughts on this? Useful or not?

useful, and not very hard

**Comment by VladislavAntonyuk** [comment]
Hey @jfversluis , is it possible to prioritize the issue? I see it was already implemented, but it wasn't merged.
It will solve MCT issue with Converters like this: https://github.com/CommunityToolkit/Maui/pull/2722

**Comment by jfversluis** [comment]
@VladislavAntonyuk unfortunately probably not until .NET 11 at this point.

**Comment by jfversluis** [comment]
Should be part of .NET 11 preview 6
